### PR TITLE
Add ability to use errorsManager in plugins

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

For now the new `errorsManager` was only exposed in the kuzzle core.
This PR allows users to use it inside their plugins by defining the custom errors in the `manifest.json` file.

For instance, if you want to use it in the `kuzzle-plugin-logger`, you should have an `errors` field in its `manifest.json`. The file should look:
```
{
    "name": "kuzzle-plugin-logger",
    "kuzzleVersion": ">=1.0.0 <2.0.0",
    "privileged": false,
    "errors": {
        "unknown_service": {
            "code": 1,
            "message": "[kuzzle-plugin-logger] Unable to find service %s",
            "class": "InternalError"
        },
        "some_error": {
            "code": 2,
            "message": "Some error occured %s",
            "class": "BadRequestError"
	},
        "some_other_error": {
            "code": 3,
            "message": "Some other error occured %s",
            "class": "ForbiddenError"
	}
    }
}

```
In the `plugins.json` (so the `plugins` domain) will be added a `kuzzle-plugin-logger` subdomain with its `errors` field filled by the one defined in the `manifest.json`.

The code of the subdomain is either `0x00` or defined in the `kuzzlerc`.
For instance, your kuzzlerc would be :

```
{
  "plugins": {
    "my-plugins": {
      "foo": "bar",
      "gordon": "freeman",
      "some": "stuff",
      "_pluginCode": {
          "my-plugins": 42
      }
    }
  }
}
```

For officials released plugins, they are now defined in the `default.config` of Kuzzle

### How should this be manually tested?

Guess what: in order to test it you need... a plugin. Yes.

For instance you can test it with the `kuzzle-plugin-logger`.
Go to the root of the plugin, and add an `errors` field in the `manifest.json` like above.

Now, if you want to throw an error, you will have to write

`context.errorsManager.throw(errorName, placeholders)`

[So here](https://github.com/kuzzleio/kuzzle-plugin-logger/blob/master/lib/index.js#L56), you could replace the line by 
`context.errorsManager.throw(unknown_service, serviceName)`

Trigger that error, and you will see the right message with the right error name

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/4797f524fb9befc165c2c6ca7f42a7be0704265c/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/eda5a980787a14b77fff69f8395f607f097f84cb/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: .travis.yml [hunk 1]
```diff
@@ -97,16 +97,21 @@ jobs:
...
       before_script:
         - npm run doc-prepare
         - npm run --prefix doc/framework clone-repos
       script:
-        - gem install typhoeus
+        - echo "Checking dead links..."
         - HYDRA_MAX_CONCURRENCY=20 npm run --prefix doc/framework dead-links
+        - echo "Checking subcodes..."
+        - npm run doc-generate-subcodes -- --output $TRAVIS_BUILD_DIR/subcodes-doc-test.md
+        - diff $TRAVIS_BUILD_DIR/subcodes-doc-test.md $TRAVIS_BUILD_DIR/doc/1/api/essentials/errors/subcodes/index.md
```
Comment (alexandrebouthinon) [inline .travis.yml]:
```suggestion
        - diff $TRAVIS_BUILD_DIR/subcodes-doc-test.md $TRAVIS_BUILD_DIR/doc/2/api/essentials/errors/subcodes/index.md
```

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/cf6516f61bf190274d7c20fcafde80835597790c/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/34b49df11709d49301c986811343c2bf2e9c4c83/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/3bebcc821dc5ed998be4eb2d202b06a26e0cb65f/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/45fdf4012885e9e96ba1166f561deea2ceeeb031/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/d2f76609ce1cd0b2ca9510315c4c91282e604c22/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/8f2ec77d1c63202e477f952a28766a76c16aff5d/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/486164a94740c2dfb47622bdbce89caa42ae4bdb/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/9d9b75c42df5768e820158c1d987cea3e95cca33/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/00cfec5332169fc6188f73c9a9155dc5dc78494b/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/a40b9f8beff20d75d4da019d509ae345858e69c5/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/a371c16f1503bec88e1a3924baac61f6fd4516c4/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: doc/generate-subcodes-doc.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/*
```
Comment (Aschen) [inline doc/generate-subcodes-doc.js]: I think this file does not belong to this PR

File: .travis.yml [hunk 3]
```diff
@@ -97,16 +97,21 @@ jobs:
         - bash .ci/sonar.sh
 
     - stage: Unit Tests & Linters
-      name: Dead link check
+      name: Documentation dead links & subcodes check
```
Comment (Aschen) [inline .travis.yml]: I think these changes does not belong to this PR

File: lib/api/core/plugins/pluginsManager.js [hunk 4]
```diff
@@ -1062,6 +1064,30 @@ class PluginsManager {
...
+      // load customs errors configuration file
+      if (plugin.manifest.raw.errors) {
+        try {
+          const pluginCode = this.config[plugin.manifest.name] && this.config[plugin.manifest.name]._pluginCode
+            ? this.config[plugin.manifest.name]._pluginCode
+            : 0x00;
+          loadPluginsErrors(plugin.manifest.raw, pluginCode);
+          this.kuzzle.log.info(`[${plugin.manifest.name}] Custom errors successfully loaded.`);
+        } catch (err) {
+          if (err.message.match(/Error configuration file/i)
```
Comment (Aschen) [inline lib/api/core/plugins/pluginsManager.js]: Why matching with a regexp ? You can throw custom errors with `assert`
```
assert(/* something */, errorsManager.getError(/* ... */));
```

File: lib/config/error-codes/index.js [hunk 5]
```diff
@@ -41,7 +41,7 @@ function checkErrors(subdomain, domain) {
 
     assert(
       !errorCodes.has(error.code),
-      `Error codes file : code ${error.code} for error '${errorName}' is not unique (domain: ${domain.code}, subdomain: ${subdomain.code}).`);
+      `Error configuration file : code ${error.code} for error '${errorName}' is not unique (domain: ${domain.code}, subdomain: ${subdomain.code}).`);
```
Comment (Aschen) [inline lib/config/error-codes/index.js]: You can throw the custom error directly here instead of throwing a string

File: lib/config/error-codes/index.js [hunk 6]
```diff
@@ -80,9 +80,11 @@ function checkSubdomains(domain) {
 
     assert(
       !subdomainCodes.has(subdomain.code),
-      `Error codes file : code ${subdomain.code} for subdomain '${subdomainName}' is not unique (domain: ${domain.code}).`);
+      `Error configuration file : code ${subdomain.code} for subdomain '${subdomainName}' is not unique (domain: ${domain.code}).`);
```
Comment (Aschen) [inline lib/config/error-codes/index.js]: Ditto

File: lib/config/error-codes/index.js [hunk 7]
```diff
@@ -130,18 +132,30 @@ function checkDomains(errorCodesFiles) {
       `Error configuration file : Field 'code' must be an integer. (domain: '${domainName}').`);
     assert(
       domain.code >= 0x00 && domain.code <= 0xFF,
-      `Invalid error configuration file : Field 'code' must be between 0 and 255. (domain: '${domainName}').`);
+      `Error configuration file : Field 'code' must be between 0 and 255. (domain: '${domainName}').`);
```
Comment (Aschen) [inline lib/config/error-codes/index.js]: Ditto

File: test/api/core/entrypoints/protocols/websocket.test.js [hunk 8]
```diff
@@ -651,6 +651,50 @@ describe('/lib/api/core/entrypoints/protocols/websocket', () => {
       should(activeConnection.socket.terminate).not.be.called();
       should(activeConnection.socket.ping).not.be.called();
     });
+
+    it('should mark a socket as dead if not available for PING', () => {
```
Comment (Aschen) [inline test/api/core/entrypoints/protocols/websocket.test.js]: These changes does not belong to this PR

File: test/api/core/httpRouter/httpRouter.test.js [hunk 9]
```diff
@@ -9,8 +9,7 @@ const
     Request,
     errors: { InternalError }
   } = require('kuzzle-common-objects'),
-  Router = require('../../../../lib/api/core/httpRouter'),
-  { HttpMessage } = require('../../../../lib/api/core/entrypoints/protocols/http');
+  Router = require('../../../../lib/api/core/httpRouter');
```
Comment (Aschen) [inline test/api/core/httpRouter/httpRouter.test.js]: These changes does not belong to this PR

File: test/api/core/notifier/notifyDocumentMDelete.test.js [hunk 10]
```diff
@@ -39,12 +39,12 @@ describe('Test: notifier.notifyDocumentMDelete', () => {
     const
       stillAlive = {
```
Comment (Aschen) [inline test/api/core/notifier/notifyDocumentMDelete.test.js]: These changes does not belong to this PR

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [pluginsManager.js#L83](https://github.com/kuzzleio/kuzzle/blob/acc2986bc7d20bc2284bb125237e1db2f3d71f92/lib/api/core/plugins/pluginsManager.js#L83): "load" expects 0 arguments, but 1 was provided. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS930)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/acc2986bc7d20bc2284bb125237e1db2f3d71f92/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [pluginsManager.js#L83](https://github.com/kuzzleio/kuzzle/blob/03d097e670fb878e07026b821bc59a2b7cb161cf/lib/api/core/plugins/pluginsManager.js#L83): "load" expects 0 arguments, but 1 was provided. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS930)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/03d097e670fb878e07026b821bc59a2b7cb161cf/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/api/core/plugins/pluginsManager.js (hunk 4 above)
Comment (Yoann-Abbes) [reply lib/api/core/plugins/pluginsManager.js]: even if i use assert, i'll need regexp to match the error message

File: lib/config/error-codes/index.js (hunk 5 above)
Comment (Yoann-Abbes) [reply lib/config/error-codes/index.js]: As you know this file is about checking if the json files are formatted correctly
Since this errors can be triggered before those checks are finished, I prefer not to use potentially bad formatted files to throw custom errors

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [pluginsManager.js#L83](https://github.com/kuzzleio/kuzzle/blob/6b2a0567a8dababbe04bc6b1b71fea95cdd5e520/lib/api/core/plugins/pluginsManager.js#L83): "load" expects 0 arguments, but 1 was provided. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS930)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/6b2a0567a8dababbe04bc6b1b71fea95cdd5e520/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [pluginsManager.js#L83](https://github.com/kuzzleio/kuzzle/blob/de074a3663392efb467230ff9acbf94d41546cee/lib/api/core/plugins/pluginsManager.js#L83): "load" expects 0 arguments, but 1 was provided. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS930)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/de074a3663392efb467230ff9acbf94d41546cee/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/config/error-codes/index.js (hunk 5 above)
Comment (benoitvidis) [reply lib/config/error-codes/index.js]: I don't think these check really apply to Kuzzle native errors. 
I agree with @Aschen , would be better to throw a native custom error and check on its subcode within the catch rather than relying on the message.

**Comment by kuzzle** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [pluginsManager.js#L83](https://github.com/kuzzleio/kuzzle/blob/012e120b40f5442e5f796b01509df664a3f20244/lib/api/core/plugins/pluginsManager.js#L83): "load" expects 0 arguments, but 1 was provided. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS930)
1. ![INFO][INFO] [pluginContext.js#L80](https://github.com/kuzzleio/kuzzle/blob/012e120b40f5442e5f796b01509df664a3f20244/lib/api/core/plugins/pluginContext.js#L80): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginContext.js#L213](https://github.com/kuzzleio/kuzzle/blob/012e120b40f5442e5f796b01509df664a3f20244/lib/api/core/plugins/pluginContext.js#L213): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/012e120b40f5442e5f796b01509df664a3f20244/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginContext.js#L80](https://github.com/kuzzleio/kuzzle/blob/db8eef93beb6b1fd3b83abb405e6715d9d2d023b/lib/api/core/plugins/pluginContext.js#L80): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginContext.js#L213](https://github.com/kuzzleio/kuzzle/blob/db8eef93beb6b1fd3b83abb405e6715d9d2d023b/lib/api/core/plugins/pluginContext.js#L213): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/db8eef93beb6b1fd3b83abb405e6715d9d2d023b/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: lib/api/core/plugins/pluginsManager.js (hunk 4 above)
Comment (Aschen) [reply lib/api/core/plugins/pluginsManager.js]: Not if you use an error from the errorManager, so you can use `error.errorName`
But I won't block this PR for this

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginContext.js#L80](https://github.com/kuzzleio/kuzzle/blob/0ac4877e71d03fc0d45d1c52b971f63b3f78cbd0/lib/api/core/plugins/pluginContext.js#L80): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginContext.js#L213](https://github.com/kuzzleio/kuzzle/blob/0ac4877e71d03fc0d45d1c52b971f63b3f78cbd0/lib/api/core/plugins/pluginContext.js#L213): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/0ac4877e71d03fc0d45d1c52b971f63b3f78cbd0/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginContext.js#L80](https://github.com/kuzzleio/kuzzle/blob/21319d7662db7e4d30d278132a469eab41b8a898/lib/api/core/plugins/pluginContext.js#L80): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginContext.js#L213](https://github.com/kuzzleio/kuzzle/blob/21319d7662db7e4d30d278132a469eab41b8a898/lib/api/core/plugins/pluginContext.js#L213): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/21319d7662db7e4d30d278132a469eab41b8a898/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginContext.js#L105](https://github.com/kuzzleio/kuzzle/blob/bd268911e3b7355960bb7df6096542a97133e03b/lib/api/core/plugins/pluginContext.js#L105): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/bd268911e3b7355960bb7df6096542a97133e03b/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginContext.js#L105](https://github.com/kuzzleio/kuzzle/blob/3e8520029dd47cbf4f7972c18af38c1373ca6dcb/lib/api/core/plugins/pluginContext.js#L105): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/3e8520029dd47cbf4f7972c18af38c1373ca6dcb/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginContext.js#L105](https://github.com/kuzzleio/kuzzle/blob/dbea4471f8f3e16df5848a88dfd8883ad607c10a/lib/api/core/plugins/pluginContext.js#L105): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/dbea4471f8f3e16df5848a88dfd8883ad607c10a/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [pluginContext.js#L105](https://github.com/kuzzleio/kuzzle/blob/81806be36faadde7666a4c613c38bcb502c60bcd/lib/api/core/plugins/pluginContext.js#L105): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [pluginsManager.js#L969](https://github.com/kuzzleio/kuzzle/blob/81806be36faadde7666a4c613c38bcb502c60bcd/lib/api/core/plugins/pluginsManager.js#L969): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
